web-screenshot/operations.py

```diff
@@ -0,0 +1,92 @@
+import base64
+import io
+import os
+import os.path
+import tempfile
+import time
+
+from PIL import Image
+from connectors.core.connector import get_logger, ConnectorError
+from integrations.crudhub import make_request, make_file_upload_request
+
+logger = get_logger('web-screenshot')
+
+
+def create_file(file_name, file_content, file_type):
+    try:
+        response = make_file_upload_request(file_name, file_content, file_type)
+        logger.info(f"File upload complete {str(response)}")
+        return response
+    except Exception as e:
+        logger.exception("An exception occurred {0}".format(e))
+        raise ConnectorError("{0}".format(e))
+
+
+def create_attachment(file_resp, desc):
+    try:
+        file_id = file_resp.get("@id")
+        file_name = file_resp.get("filename")
+        response = make_request('/api/3/attachments', 'POST',
+                                {'name': file_name, 'file': file_id, 'description': desc})
+        logger.info(f"File attachment complete {str(response)}")
+        return response
+    except Exception as e:
+        logger.exception("An exception occurred {0}".format(e))
+        raise ConnectorError("{0}".format(e))
+
+
+def take_screenshot(config, params):
+    binary = config.get("path").strip()
+    url = params.get("url").strip()
+    if not url.startswith('http'):
+        url = f"http://{url}"
+    width = config.get("width")
+    height = config.get("height")
+    t_width = config.get("t_width")
+    t_height = config.get("t_height")
+    tmp_name = f"{next(tempfile._get_candidate_names())}.png"
+    tmp_path = f"/tmp/{tmp_name}"
+    if "chrome" not in binary.lower():
+        raise ConnectorError("Unable to find chrome binary")
+
+    cmd_parameter = ["--ignore-certificate-errors",
+                     # "--no-sandbox",
+                     "--incognito",
+                     "--disable-gpu",
+                     "--allow-running-insecure-content",
+                     "--disable-web-security",
+                     "--headless",
+                     "--virtual-time-budget=10000",
+                     f"--window-size={width},{height}",
+                     f"--screenshot=\"{tmp_path}\"",
+                     f"\"{url}\""]
+    cmd_parameter = " ".join(cmd_parameter)
+    try:
+        os.system(f"{binary} {cmd_parameter}")
```

Comment:
subprocess /popen would probably a better choice since it allows you use stdout/stderr so a better error handling